### ISSUE TITLE
misleading skip message in test_strcred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3170]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3170|
|Type|defect|
|Created|2008-04-07 14:16:18Z|

If _any_ of `pwd`, `spwd`, or `crypto` is unavailable, the skip message for several tests in `test_strcred` is _pwd and spwd are both unavailable_.

Attachments:

* [3170.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/214/2145c3e7ce74ef291f7254708749ad33403111fc/d346aad7a2218a226b2d9d92ebba82db4a08795d.diff) (905 bytes) - added by cyli on 2010-05-23 22:06:48Z - Fix the test_strcred.py module unavailability message
* [3170.2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/214/2145c3e7ce74ef291f7254708749ad33403111fc/1ae77d12b52a7eae0f4384edceded1b518f8436c.diff) (795 bytes) - added by cyli on 2010-05-24 21:46:16Z - Shorter test

<details><summary>Searchable metadata</summary>

```
trac-id__3170 3170
type__defect defect
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1207577778000000 1207577778000000
changetime__1275246082000000 1275246082000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|TimAllen set owner to @cyli|
|-|-|

Thanks for supplying a patch!

The skip message needs to accurately reflect the problem, but it doesn't necessarily need to be properly formatted English. I think the code could be simpler if you aimed at producing a message like:
```
Required modules are unavailable: pwd, spwd, crypt
```

Thanks again for your efforts!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph commented|
|-|-|

Replying to [TimAllen](#note_3):
> Thanks for supplying a patch!
> 
> The skip message needs to accurately reflect the problem, but it doesn't necessarily need to be properly formatted English. I think the code could be simpler ...

If I may interject a meta-review here...

I do appreciate that you're doing a code-review, and I can appreciate the desire for test code to be simpler, but:

  1. this review doesn't point out any test-coverage, doc-coverage, or coding-standard compliance issues
  2. it asks the contributor to do more work
  3. it recommends that the error message be made _harder_ to read, despite the work of the more properly-formatted english message already having been done, which is clearly a pretty subjective opinion
  4. it leaves the ticket in kind of a limbo state: are you saying "do these changes, then merge it", or "please submit for re-review" or "here are some ideas I had, but I'm not really changing the state of the ticket"?

Especially for simple contributions like this, I think it's important that we avoid doing code reviews that create more work or require the contributor to deal with arbitrary delays in getting their work accepted.  If the change is acceptable based on our coding standards and deals with the issue described in the ticket properly, it should generally be accepted.  It's hard enough to meet that bar without adding arbitrary other qualifications that vary on a per-reviewer basis!

Sorry to pick on you personally, TimAllen, this is just representative of a somewhat worrying trend I've seen (maybe even in some of my own reviews).  Reviewers should strive to be as objective and consistent as possible, to avoid sending mixed messages to contributors.

I'll let cyli decide if she actually wants to change the code or simply submit for re-review, but please consider these issues when doing future reviews.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli removed owner|
|-|-|

Modified so that it's slightly shorter

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @cyli|
|-|-|

 * Please update the copyright of the test file
 * There is a trailing whitespace in the diff
 * It needs a NEWS fragment.

Please apply to trunk once done.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli set status to `closed`|
|-|-|

(In [29103]) Applied patch 3170.2.diff from ticket 3170: changed skip message in test_strcred

Author: cyli 
Reviewer: TimAllen, therve 
Fixes: [#3170](https://github.com/twisted/twisted/issues/3170)

Changed the skip message in test_strcred to note which modules are unavailable, and added a misc news file.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
